### PR TITLE
baremetalds-packet: Change CLUSTER_TYPE check to a regex match

### PR DESCRIPTION
**Pull request by lranjbar**
I'm looking to update the assisted installer CLUSTER_TYPE to match our qouta type `packet-edge`. I added in equinix if we feel like renaming things in the future.

/assign @andfasano @derekhiggins

### CODE REVIEW

**Comment by lranjbar** [comment]
This is all related to #24999 where I am trying to move the `cluster_profile` definition to the workflows.

File: ci-operator/step-registry/baremetalds/packet/check/baremetalds-packet-check-commands.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ PACKET_AUTH_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/packet-auth-token")
 SLACK_AUTH_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/slackhook")
 
 # Initial check
-if [ "${CLUSTER_TYPE}" != "packet" ] ; then
+if [[ "${CLUSTER_TYPE}" =~ ^packet.*$|^equinix.*$ ]] ; then
```
Comment (derekhiggins) [inline ci-operator/step-registry/baremetalds/packet/check/baremetalds-packet-check-commands.sh:14]: should be negated
`[[ ! "${CL....`

File: ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh [hunk 2]
```diff
@@ -30,7 +30,7 @@ cat > packet-setup.yaml <<-EOF
   - name: check cluster type
     fail:
       msg: "Unsupported CLUSTER_TYPE '{{ cluster_type }}'"
-    when: cluster_type != "packet"
+    when: "cluster_type is not regex('^packet.*$|^equinix.*$')"
```
Comment (derekhiggins) [inline ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh:33]: See line 100 below do we also need a check there?

**Review by derekhiggins — COMMENTED** [review]
There seems to be another check on CLUSTER_TYPE in e2e/test/baremetalds-e2e-test-commands.sh

**Pull request merged**